### DISCUSSION TITLE
Wipe Original File

### DISCUSSION THREAD

**Discussion by OA971** [FiloSottile/age#470] [General]
Hi, 
Is it a good idea to add an option that wipes the original file, whether it's the source file or the encrypted file?

**Comment by alerque** [comment]
This doesn't sound like a good plan to me. I understand you would want it behind an option flag, but even so one does not usually expect any sort of compression/encryption/bundling app to delete sources. There are too many things that could go wrong and the user is expected to test the situation before moving forward. Even having it as an option kind of invites trouble: "YOUR BUG ATE MY DATA". But beyond that, the concept of "wiping" is not universal or easy. Some physical mediums have a way to do this semi-securely, but many do not and even the ones that do do not have a consistent API or way to accomplish it across platforms. Instead I think this project should (if anything) document the need for wiping and point people towards platform & hardware specific tools to build out their full workflow.

**Reply by OA971** [reply]
You can execute sdelete.
Or give me an option for a command line after encryption or decryption instead of executing it manually.
https://learn.microsoft.com/en-us/sysinternals/downloads/sdelete

Well, it might not be liked by some people but others could find it helpful.

**Comment by FiloSottile** [comment]
This is out of scope for age. Securely deleting data is an extremely nuanced task that depends on the filesystem, operating system, and hardware. age can't possibly do it safely in the general case. Also, age is not meant to be a swiss army knife, but a focused tool that does encryption. If you want to do encryption and deletion, you can combine age and a tool for deletion.
